lab5-8/ui.py

```diff
@@ -0,0 +1,80 @@
+from domain.grammar import Grammar
+from domain.parser import Parser
+from domain.tree import Tree
+
+
+class UI:
+
+    def __init__(self):
+        self.grammar = None
+        self.parser = None
+
+    def run(self):
+        while True:
+            print(">>")
+            cmd = input()
+            if cmd == "g1":
+                self.evaluateG1()
+            elif cmd == "g2":
+                self.evaluateG2()
+
+    def readG1(self):
+        self.g1 = Grammar.fromFile('g1.txt')
+        print('Read g1')
+
+    def readG2(self):
+        self.g2 = Grammar.fromFile('g2.txt')
+        print('Read g2')
+
+    def readSequence(self, fname):
+        sequence = ""
+        with open(fname, 'r') as fin:
+            for line in fin.readlines():
+                sequence += line.strip() + " "
+        return sequence.strip()
+
+    def evaluateG1(self):
```

Comment:
only the grammar file and the sequence file differ between this method and `evaluateG2`, could have been instead a method with those 2 as parameters